src/Tabs/Tabs.js

```diff
@@ -0,0 +1,266 @@
+    handleSideScroll = () => {
+        this.toggleScrollButtonVisibility();
+    };
+
+    scrollLeft = () => {
+        const { scrollBox, tabs } = this.nodes;
+        const offsetLeft = scrollBox.scrollLeft - scrollBox.offsetWidth;
+        const targetTab = offsetLeft <= 0 ? tabs[0] : this.getTabAtOffsetLeft(offsetLeft);
+        this.scrollToTab(targetTab);
+    };
+
+    scrollRight = () => {
+        const { scrollBox, scrollArea, tabs } = this.nodes;
+        const offsetLeft = scrollBox.scrollLeft + scrollBox.offsetWidth * 2;
+        const atEnd = scrollArea.offsetWidth <= offsetLeft;
+        const targetTab = atEnd
+            ? tabs[tabs.length - 1]
+            : this.getTabAtOffsetLeft(offsetLeft);
+        this.scrollToTab(targetTab);
+    };
+
+    showTabIndicator = () => {
+        this.setState({ showTabIndicator: true });
+    };
+
+    // Methods
+    getTabAtOffsetLeft(offsetLeft) {
+        return this.nodes.tabs.find(
+            tab =>
+                offsetLeft >= tab.offsetLeft &&
+                offsetLeft <= tab.offsetLeft + tab.offsetWidth
+        );
+    }
+
+    scrollRequiredToReachActiveTab() {
+        const { scrollBox } = this.nodes;
+        const tab = this.getActiveTabRef();
+        const shouldScrollRight =
+            tab.offsetLeft - scrollBox.scrollLeft + tab.offsetWidth >
+            scrollBox.offsetWidth;
+        const shouldScrollLeft = tab.offsetLeft < scrollBox.scrollLeft;
+        return shouldScrollRight || shouldScrollLeft;
+    }
+
+    scrollToTab(tab, scrollParams = {}) {
+        animatedScrollTo({
+            to: tab,
+            scrollBox: this.nodes.scrollBox,
+            direction: 'horizontal',
+            ...scrollParams,
+        });
+    }
+
+    toggleScrollButtonVisibility() {
+        const { scrollBox, scrollArea } = this.nodes;
+        const scrolledToStart = scrollBox.scrollLeft === 0;
+        const scrolledToEnd =
+            scrollBox.scrollLeft + scrollBox.offsetWidth === scrollArea.offsetWidth;
+
+        if (
+            this.state.scrolledToStart !== scrolledToStart ||
+            this.state.scrolledToEnd !== scrolledToEnd
+        ) {
+            this.setState({
+                scrolledToStart,
+                scrolledToEnd,
+            });
+        }
+    }
+
+    getAdditionalTabProps(index) {
+        const { stackedTabs, activeTabIndex } = this.props;
+        return {
+            stacked: stackedTabs,
+            active: index === activeTabIndex,
+            addTabRef: this.addTabRef,
+        };
+    }
+
+    renderTabItems() {
+        const { tabItems } = this.props;
+        return tabItems.map((tab, index) => (
+            <Tab key={`tab-${index}`} {...this.getAdditionalTabProps(index)} {...tab} />
+        ));
+    }
+
+    renderChildren() {
+        const { children } = this.props;
+        return Children.map(children, (child, index) =>
+            cloneElement(child, this.getAdditionalTabProps(index))
+        );
```

Comment:
So this is related to my previous comment about required prop-types in the Tab component. If you use the `Tabs` like this:
```jsx
<Tabs>
    <Tab />
    <Tab />
</Tabs>
```
Then each `Tab` will be patched with some additional properties when the `Tabs` renders it. I'm pretty sure that the way I've done this is pretty compliant to React _best-practices_ but because the prop-types package is complaining if one of the added properties is marked as required, I'm not so sure.

Could you think of another type of solution?

Reply:
Not sure, I cannot reproduce the problem to debug it?